crates/hyfetch/src/bin/hyfetch.rs

```diff
@@ -0,0 +1,1034 @@
+                }
+                printc(line.join("  "), color_mode).context("failed to print test ascii line")?;
+            }
+        }
+
+        loop {
+            writeln!(
+                io::stdout(),
+                "\nWhich brightness level looks the best? (Default: {default:.0}% for \
+                 {light_dark} mode)",
+                default = f32::from(default_lightness) * 100.0,
+                light_dark = theme.as_ref()
+            )
+            .context("failed to write prompt to stdout")?;
+            let lightness = input(Some("> "))
+                .context("failed to read input")?
+                .trim()
+                .to_lowercase();
+
+            match parse_lightness(lightness, default_lightness) {
+                Ok(lightness) => {
+                    return Ok(lightness);
+                },
+                Err(err) => {
+                    debug!(%err, "could not parse lightness");
+                    printc(
+                        "&cUnable to parse lightness value, please enter a lightness value such \
+                         as 45%, .45, or 45",
+                        color_mode,
+                    )
+                    .context("failed to print message")?;
+                },
+            }
+        }
+    };
+
+    fn parse_lightness(lightness: String, default: Lightness) -> Result<Lightness> {
+        if lightness.is_empty() || ["unset", "none"].contains(&&*lightness) {
+            return Ok(default);
+        }
+
+        let lightness = if let Some(lightness) = lightness.strip_suffix('%') {
+            let lightness: RangedU8<0, 100> = lightness.parse()?;
+            lightness.get() as f32 / 100.0
+        } else {
+            match lightness.parse::<RangedU8<0, 100>>() {
+                Ok(lightness) => lightness.get() as f32 / 100.0,
+                Err(_) => lightness.parse::<f32>()?,
+            }
+        };
+
+        Ok(Lightness::new(lightness)?)
+    }
+
+    let lightness = select_lightness().context("failed to select lightness")?;
+    debug!(?lightness, "selected lightness");
+    let color_profile = color_profile.with_lightness_adaptive(lightness, theme);
+    update_title(
+        &mut title,
+        &mut option_counter,
+        "Selected brightness",
+        &format!("{lightness:.2}", lightness = f32::from(lightness)),
+    );
+
+    //////////////////////////////
+    // 5. Color arrangement
+
+    let color_align: ColorAlignment;
+
+    // Calculate amount of row/column that can be displayed on screen
+    let (ascii_per_row, ascii_rows) = {
+        let (Width(term_w), Height(term_h)) =
+            terminal_size().context("failed to get terminal size")?;
+        let ascii_per_row = cmp::max(1, term_w / (u16::from(asc_width) + 2));
+        let ascii_per_row =
+            u8::try_from(ascii_per_row).expect("`ascii_per_row` should fit in `u8`");
+        let ascii_rows = cmp::max(1, (term_h - 8) / (u16::from(asc_lines) + 1));
```

Comment:
When the terminal is only a few lines in height, this will underflow (unsigned), and then later panic on the u8 conversion.